saleor/graphql/core/mutations.py

```diff
@@ -158,12 +160,26 @@ def _get_node_by_pk(
         Whether by using the provided query set object or by calling type's get_node().
         """
         if qs is not None:
+            if str(graphene_type) == "Order":
+                return cls._get_order_node_by_pk(pk, qs)
             return qs.filter(pk=pk).first()
         get_node = getattr(graphene_type, "get_node", None)
         if get_node:
             return get_node(info, pk)
         return None
 
+    @classmethod
+    def _get_order_node_by_pk(cls, pk: Union[int, str], qs):
+        # This is temporary method that allows fetching orders with use of
+        # new and old id.
+        lookup = Q(pk=pk)
+        if pk is not None:
+            try:
+                UUID(str(pk))
+            except ValueError:
+                lookup = Q(number=pk) & Q(use_old_id=True)
```

Comment:
Should we check `use_old_id` flag? 
```suggestion
                lookup = Q(number=pk) | (Q(use_old_id=True) & Q(use_old_id=True)
```